src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -358,10 +361,19 @@ def get_master_zone_ep(self, realm, zonegroup=None):
 
     def get_master_zonegroup(self, realm):
         try:
-            ze = ZoneEnv(self.env, realm=realm)
-            params = ['zonegroup', 'get']
-            return RGWAdminJSONCmd(ze).run(params)
-        except RGWAMCmdRunException:
+            # Fetch the realm period
+            realm_period = self.get(realm)
+            master_zonegroup_id = realm_period['master_zonegroup']
+            zonegroups = realm_period['period_map']['zonegroups']
+
+            # Find the master zonegroup in the realm period data
+            for zonegroup in zonegroups:
+                if zonegroup['id'] == master_zonegroup_id:
+                    return zonegroup
+            return None
```

Comment:
This pattern is repeated dozens of times. We should probably add a `find()` helper:
```suggestion
            # Find the master zonegroup in the realm period data
            return find_item(zonegroups, lambda i: i['id'] == master_zonegroup_id, defalt=None)
```
Funny thing is that `next` does exactly this:
```suggestion
            # Find the master zonegroup in the realm period data
            return next((zg for zg in zonegroups if zg['id'] == master_zonegroup_id), None)
```